ui/helper/clicker.go

```diff
@@ -0,0 +1,80 @@
+package helper
+
+import (
+	"time"
+
+	"gioui.org/f32"
+	"gioui.org/gesture"
+	"gioui.org/layout"
+	"gioui.org/op"
+)
+
+type (
+	clickItem struct {
+		position f32.Point
+		time     time.Time
+	}
+
+	Clicker struct {
+		click      gesture.Click
+		clicks     int
+		prevClicks int
+		history    []clickItem
+	}
+)
+
+func NewClicker() Clicker {
+	return Clicker{
+		history: []clickItem{},
+	}
+}
+
+func (c *Clicker) Clicked(ctx *layout.Context) bool {
```

Comment:
Why use this instead of the widget's inbuilt clicked function?

Reply:
we need custom made widgets that are clickable. There's no inbuilt way of doing that. This helper gives us the feature of making our own widgets clickable

Reply:
Buttons are clickable, why not use buttons?